packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -162,12 +220,13 @@ function trimOldestLines(
 function registerTabChangeCallback(
     session: LiveTailSession,
     registry: LiveTailSessionRegistry,
-    document: vscode.TextDocument
+    document: vscode.TextDocument,
+    timer: NodeJS.Timer
```

Comment:
Did you check if our `Timeout` module is useful? https://github.com/aws/aws-toolkit-vscode/blob/34b695a91cc0568734c980b1931ac9952c37869c/packages/core/src/shared/utilities/timeoutUtils.ts#L41

It supports cancellations and other features, so mentioning it in case your needs expand later.